internal/provider/group_resource_test.go

```diff
@@ -0,0 +1,229 @@
+package provider
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseHttpResponse(t *testing.T) {
+	t.Run("Basic Test", func(t *testing.T) {
+		g := groupResourceModel{
+			Name: types.StringValue("group1"),
+			URL:  types.StringValue("/api/v2/groups/24/"),
+		}
+		body := []byte(`{"name": "group1", "url": "/api/v2/groups/24/", "description": ""}`)
+		err := g.ParseHttpResponse(body)
+		assert.NoError(t, err)
+	})
+	t.Run("Test with variables", func(t *testing.T) {
+		g := groupResourceModel{
+			Name: types.StringValue("group1"),
+			URL:  types.StringValue("/api/v2/groups/24/"),
+		}
+		body := []byte(`{"name": "group1", "url": "/api/v2/groups/24/", "description": "", "variables": "{\"ansible_network_os\":\"ios\"}"}`)
+		err := g.ParseHttpResponse(body)
+		assert.NoError(t, err)
+	})
+}
+
+func TestCreateRequestBody(t *testing.T) {
+	t.Run("Basic Test", func(t *testing.T) {
+		g := groupResourceModel{
+			InventoryId: basetypes.NewInt64Value(1),
+			Name:        types.StringValue("group1"),
+			URL:         types.StringValue("/api/v2/groups/24/"),
+		}
+		body := []byte(`{"inventory": 1, "name": "group1", "url": "/api/v2/groups/24/"}`)
+		result, diags := g.CreateRequestBody()
+		if diags.HasError() {
+			t.Fatal(diags.Errors())
+		}
+		test, err := DeepEqualJSONByte(body, result)
```

Comment:
I'm guessing the reason for this is because two JSON objects may be functionally equivalent, but may serialize to a different byte sequence because of key ordering. If you can guarantee the order of keys, you could just do a simple byte comparison, but it's probably not worth it. It's easier to just unmarshal and then compare the resulting structs.

Reply:
I just noticed you added testify in this PR. If we're using testify, there's already a function that compares JSON strings: https://pkg.go.dev/github.com/stretchr/testify/require?utm_source=godoc#JSONEq